0.15.2 release notes, manpage and version bump

Sjors opened the pull request:
Semi manual based on [commits since v0.15.1](https://github.com/bitcoin/bitcoin/compare/v0.15.1...0.15), so please sanity check.

hebasto commented on doc/release-notes.md:
Do you mean:
```
- #11647 `2eea279` Make AuthServiceProxy.\_batch method usable (Russell Yanofsky)
- #11647 `305f768` Limit AuthServiceProxyWrapper.\_\_getattr\_\_ wrapping (Russell Yanofsky)
```

Sjors replied:
Thanks, added escape characters.

Sjors commented:
Removed translation credits, because they weren't updated for this release.

Sjors committed:
doc: Update manpages for 0.15.2

laanwj commented:
I think it'd make sense to combine this with #14263 and a bump-version commit

fanquake commented on doc/release-notes.md:
#11662 is a backport of #11554 from TheBlueMatt

fanquake commented on doc/release-notes.md:
#12032 is a backport of #11847 from sdaftuar

Sjors committed:
doc: 0.15.2 release notes

Sjors committed:
build: Bump version to 0.15.2

Sjors commented:
* updated pull request numbers to reference originals rather than the backport. This is especially useful for #11647 which was one backport PR multiple PR's
* added missing authors @fanquake pointed out
* added RPC and Tests category
* added a version update commit
* moved manpage commit from #14263

Sjors commented:
From the manpage PR:
> I used the generator script from master and fixed the version number after running the script.
>
> Notable changes:
> * script verification threads changed from 6 to 4
> * maximum BIP141 block weight default changed from 3000000 to 3996000
> * blockmaxsize deprecated

laanwj commented:
ACK 391b2cf0250de44afcaeaaf977e3ef8ec5c4d55f

Thank you for doing this!
